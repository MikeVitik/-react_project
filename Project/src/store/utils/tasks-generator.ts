import { TASK_TIME, TOTAL_MILLISECONDS_IN_DAY } from "../const";
import { SerialiazableTask } from "../slicies/tasks-slice";

export function generateTasks(
  currentDate = new Date(),
  seed = 1
): SerialiazableTask[] {
  function random(max: number) {
    var x = Math.sin(seed++) * 10000;
    return Math.floor(max * (x - Math.floor(x)));
  }

  const result: SerialiazableTask[] = [];
  let id = -1;
  for (let i = 0; i < 7 * 3; i++) {
    const completedPomodoro = (random(4) + 1) * TASK_TIME;
    result.push({
      name: "Generated task " + i,
      id,
      startDateString: currentDate.toUTCString(),
      isCompleted: true,
      pomodoroCount: 3,
      completedPomodoro,
      workTime: Math.floor(
        completedPomodoro + completedPomodoro * (random(30) / 100)
      ),
      pauseCount: random(4),
      pauseTime: random(20 * 1000),
    });
    currentDate = new Date(currentDate.getTime() - TOTAL_MILLISECONDS_IN_DAY);
    id = id - 1;
  }
  return result.reverse();
}
