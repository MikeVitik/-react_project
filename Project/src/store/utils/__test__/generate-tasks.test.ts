import { generateTasks } from "../tasks-generator";

describe("generateTasks", () => {
  it("should generate task for 2 weeks", () => {
    const tasks = generateTasks(new Date(2024, 1, 20));
    expect(tasks.length).toBe(21);
    expect(tasks.filter(({ id }) => id < 0).length).toBe(21);
    expect(tasks).toMatchSnapshot();
  });
});
