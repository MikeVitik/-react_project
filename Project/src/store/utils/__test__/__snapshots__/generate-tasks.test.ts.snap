// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTasks should generate task for 2 weeks 1`] = `
Array [
  Object {
    "completedPomodoro": 20000,
    "id": -21,
    "isCompleted": true,
    "name": "Generated task 20",
    "pauseCount": 2,
    "pauseTime": 18064,
    "pomodoroCount": 3,
    "startDateString": "Tue, 30 Jan 2024 21:00:00 GMT",
    "workTime": 21600,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -20,
    "isCompleted": true,
    "name": "Generated task 19",
    "pauseCount": 3,
    "pauseTime": 2269,
    "pomodoroCount": 3,
    "startDateString": "Wed, 31 Jan 2024 21:00:00 GMT",
    "workTime": 24600,
  },
  Object {
    "completedPomodoro": 40000,
    "id": -19,
    "isCompleted": true,
    "name": "Generated task 18",
    "pauseCount": 0,
    "pauseTime": 1527,
    "pomodoroCount": 3,
    "startDateString": "Thu, 01 Feb 2024 21:00:00 GMT",
    "workTime": 46400,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -18,
    "isCompleted": true,
    "name": "Generated task 17",
    "pauseCount": 2,
    "pauseTime": 4672,
    "pomodoroCount": 3,
    "startDateString": "Fri, 02 Feb 2024 21:00:00 GMT",
    "workTime": 25400,
  },
  Object {
    "completedPomodoro": 40000,
    "id": -17,
    "isCompleted": true,
    "name": "Generated task 16",
    "pauseCount": 3,
    "pauseTime": 14463,
    "pomodoroCount": 3,
    "startDateString": "Sat, 03 Feb 2024 21:00:00 GMT",
    "workTime": 45600,
  },
  Object {
    "completedPomodoro": 80000,
    "id": -16,
    "isCompleted": true,
    "name": "Generated task 15",
    "pauseCount": 2,
    "pauseTime": 5207,
    "pomodoroCount": 3,
    "startDateString": "Sun, 04 Feb 2024 21:00:00 GMT",
    "workTime": 84000,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -15,
    "isCompleted": true,
    "name": "Generated task 14",
    "pauseCount": 1,
    "pauseTime": 17875,
    "pomodoroCount": 3,
    "startDateString": "Mon, 05 Feb 2024 21:00:00 GMT",
    "workTime": 72600,
  },
  Object {
    "completedPomodoro": 40000,
    "id": -14,
    "isCompleted": true,
    "name": "Generated task 13",
    "pauseCount": 1,
    "pauseTime": 9799,
    "pomodoroCount": 3,
    "startDateString": "Tue, 06 Feb 2024 21:00:00 GMT",
    "workTime": 41200,
  },
  Object {
    "completedPomodoro": 40000,
    "id": -13,
    "isCompleted": true,
    "name": "Generated task 12",
    "pauseCount": 1,
    "pauseTime": 5518,
    "pomodoroCount": 3,
    "startDateString": "Wed, 07 Feb 2024 21:00:00 GMT",
    "workTime": 42800,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -12,
    "isCompleted": true,
    "name": "Generated task 11",
    "pauseCount": 2,
    "pauseTime": 9067,
    "pomodoroCount": 3,
    "startDateString": "Thu, 08 Feb 2024 21:00:00 GMT",
    "workTime": 25200,
  },
  Object {
    "completedPomodoro": 80000,
    "id": -11,
    "isCompleted": true,
    "name": "Generated task 10",
    "pauseCount": 1,
    "pauseTime": 385,
    "pomodoroCount": 3,
    "startDateString": "Fri, 09 Feb 2024 21:00:00 GMT",
    "workTime": 98400,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -10,
    "isCompleted": true,
    "name": "Generated task 9",
    "pauseCount": 3,
    "pauseTime": 2632,
    "pomodoroCount": 3,
    "startDateString": "Sat, 10 Feb 2024 21:00:00 GMT",
    "workTime": 72000,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -9,
    "isCompleted": true,
    "name": "Generated task 8",
    "pauseCount": 0,
    "pauseTime": 4229,
    "pomodoroCount": 3,
    "startDateString": "Sun, 11 Feb 2024 21:00:00 GMT",
    "workTime": 24800,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -8,
    "isCompleted": true,
    "name": "Generated task 7",
    "pauseCount": 2,
    "pauseTime": 5336,
    "pomodoroCount": 3,
    "startDateString": "Mon, 12 Feb 2024 21:00:00 GMT",
    "workTime": 72000,
  },
  Object {
    "completedPomodoro": 40000,
    "id": -7,
    "isCompleted": true,
    "name": "Generated task 6",
    "pauseCount": 3,
    "pauseTime": 1157,
    "pomodoroCount": 3,
    "startDateString": "Tue, 13 Feb 2024 21:00:00 GMT",
    "workTime": 46800,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -6,
    "isCompleted": true,
    "name": "Generated task 5",
    "pauseCount": 3,
    "pauseTime": 4327,
    "pomodoroCount": 3,
    "startDateString": "Wed, 14 Feb 2024 21:00:00 GMT",
    "workTime": 68400,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -5,
    "isCompleted": true,
    "name": "Generated task 4",
    "pauseCount": 3,
    "pauseTime": 9050,
    "pomodoroCount": 3,
    "startDateString": "Thu, 15 Feb 2024 21:00:00 GMT",
    "workTime": 20600,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -4,
    "isCompleted": true,
    "name": "Generated task 3",
    "pauseCount": 3,
    "pauseTime": 19336,
    "pomodoroCount": 3,
    "startDateString": "Fri, 16 Feb 2024 21:00:00 GMT",
    "workTime": 61200,
  },
  Object {
    "completedPomodoro": 20000,
    "id": -3,
    "isCompleted": true,
    "name": "Generated task 2",
    "pauseCount": 0,
    "pauseTime": 5416,
    "pomodoroCount": 3,
    "startDateString": "Sat, 17 Feb 2024 21:00:00 GMT",
    "workTime": 24600,
  },
  Object {
    "completedPomodoro": 80000,
    "id": -2,
    "isCompleted": true,
    "name": "Generated task 1",
    "pauseCount": 3,
    "pauseTime": 11649,
    "pomodoroCount": 3,
    "startDateString": "Sun, 18 Feb 2024 21:00:00 GMT",
    "workTime": 100000,
  },
  Object {
    "completedPomodoro": 60000,
    "id": -1,
    "isCompleted": true,
    "name": "Generated task 0",
    "pauseCount": 0,
    "pauseTime": 19500,
    "pomodoroCount": 3,
    "startDateString": "Mon, 19 Feb 2024 21:00:00 GMT",
    "workTime": 77400,
  },
]
`;
